rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,204 @@
+
+### **What system will work for Razorpay?**
+We should consider few cases that will make the system flexible for us. Few things to consider here are the **user needs**.
+
+As a merchant (user) do they want the interface to be spacious with large fonts and limited number of actions? Absolutely not, we need something that doesn't look *too-dense* or *too-spacious*. Therefore we'll consider **`4pt base unit`** which will give us the flexibility to scale the UI in multiples of `x4`.
+
+We also need to focus on the number of variables we'll have in our scale. These are the variables/tokens developers will use to while defining a grid layout.
+
+### **How to apply a spatial system?**
+Applying a system can come in the form of `padding`, `margins`, `width` & `height`. So there are two things we can control here *main sizing* & *internal element sizing*.
+
+- **Element-first (strict element sizing):** In this type the size takes the priority. For examples *buttons*, *form inputs*, *etc*.
+![Element-First Spacing](./images/spatial/element-first.png)
+
+- **Content first (strict internal padding):** This is to be used when the content is dynamic or less predictable. Therefore, we will enforce the strict internal padding to those components. Simply put, this is the spacing around the elements.
+![Content-First Spacing](./images/spatial/content-first.png)
+
+### **Spatial Scale**
+We'll follow a **8pt** linear scale for the elements with the **4pt** half-step for spacing smaller content (icons, etc.) and typography (line-heights).
+
+This is how the scale will look like:
+
+| Alias      | Value |
+|------------|-------|
+| $spacer-0  | 0px   |
+| $spacer-1  | 4px   |
+| $spacer-2  | 8px   |
+| $spacer-3  | 12px  |
+| $spacer-4  | 16px  |
+| $spacer-5  | 20px  |
+| $spacer-6  | 24px  |
+| $spacer-7  | 32px  |
+| $spacer-8  | 40px  |
+| $spacer-9  | 48px  |
+| $spacer-10 | 56px  |
+
+>These are mixed values, it contains multiple of `4` as well as `8`.
+
+### **Grids**
+The grid system helps align page elements based on sequenced columns and rows. Basically it helps you arrange your structure into structured manner. Here we'll look into 2D organisation of information.
+
+### **Column Grid**
+As the name suggests, this type of grid will help you organise content into evenly spaced vertical columns. The spacer which divide this grid is called as `gutter`. We'll use a **12 column grid** as it helps divide the area into *halfs*, *thirds*, *fourths*, *sixths*.
+
+Here we'll have 3 types of column grids, these will be used depending on the use case. All of them are following the same maximum width of `1280px`.
+1. **Default:** This has a gutter of `32px` & will be used on the internal dashboard.
+![Default Column-Grid](./images/spatial/col-grid-default.png)
+
+2. **Condensed:** This has a gutter of `16px` & will be used on mobile layouts *(only for dashboards)*.
+![Condensed Column-Grid](./images/spatial/col-grid-condensed.png)
```

Comment:
Shouldn't the width over should show `360px` which is the normal layout? because `375` is odd and you have shown what to do in that case below. Also the grid won't be equally divided on `375px` but the diagram shows equal distribution which is kind of misleading